internal/controller/storageclient_controller.go

```diff
@@ -367,7 +367,7 @@ func (r *StorageClientReconciler) newExternalClusterClient() (*providerClient.OC
 	ocsProviderClient, err := providerClient.NewProviderClient(
 		r.ctx, r.storageClient.Spec.StorageProviderEndpoint, time.Second*10)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create a new provider client: %v", err)
+		return nil, fmt.Errorf("failed to create a new provider client while connecting to %v: %v", r.storageClient.Spec.StorageProviderEndpoint, err)
```

Comment:
we need this at all places where we are creating new client, like, in claim controller and in heartbeat as well.

Reply:
Updated

Reply:
Did you check other places as well? I mentioned what I remembered.

Reply:
I checked where `NewProviderClient` is created